WordPress/src/main/java/org/wordpress/android/WordPress.java

```diff
@@ -195,35 +216,24 @@ public void onCreate() {
                 .appContextModule(new AppContextModule(getApplicationContext()))
                 .build();
         component().inject(this);
+        mDispatcher.register(this);
 
         // Init static fields from dagger injected singletons, for legacy Actions/Utils
         sRequestQueue = mRequestQueue;
         sImageLoader = mImageLoader;
         sOAuthAuthenticator = mOAuthAuthenticator;
 
-        // Migrate access token AccountStore
-        if (!mAccountStore.hasAccessToken()) {
-            // it will take some time to update the access token in the AccountStore if it was migrated
-            // so it will be set to the migrated token
-            String migratedToken = WPLegacyMigrationUtils.migrateAccessTokenToAccountStore(this, mDispatcher);
-            if (!TextUtils.isEmpty(migratedToken)) {
-                AppPrefs.setAccessTokenMigrated(true);
-                mDispatcher.dispatch(AccountActionBuilder.newFetchAccountAction());
-                mDispatcher.dispatch(AccountActionBuilder.newFetchSettingsAction());
-                mDispatcher.dispatch(SiteActionBuilder.newFetchSitesAction());
+        if (!AppPrefs.wasAccessTokenMigrated() || !AppPrefs.isSelfHostedSitesMigratedToFluxC()) {
```

Comment:
We should make sure there is something to migrate, we don't want to show this if it's first time signin.